exporters/prometheus/exporter.go

```diff
@@ -192,7 +192,7 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 
 		if !c.disableScopeInfo {
 			scopeInfo, err := c.scopeInfo(scopeMetrics.Scope)
-			if err == errScopeInvalid {
+			if errors.Is(err, errScopeInvalid) {
```

Comment:
Have you checked the performance overhead? While I have nothing against using `errors.Is` and `errors.As` in business logic I am a uncertain if we to use it here (it looks like a hot path). As far as I remember it uses some kind of reflection (`internal/reflectlight`).

CC @dashpole

Reply:
I created https://github.com/open-telemetry/opentelemetry-go/issues/5551